core/src/smartcontracts/wasm.rs

```diff
@@ -1241,28 +1392,44 @@ impl<'wrld> import_traits::ExecuteOperations<state::executor::ValidateQuery<'wrl
     #[codec::wrap]
     fn execute_query(
         query_request: SmartContractQueryRequest,
-        state: &state::executor::ValidateQuery<'wrld>,
+        state: &mut state::executor::ValidateQuery<'wrld>,
     ) -> Result<BatchedResponse<Value>, ValidationFail> {
         debug!(%query_request, "Executing as executor");
 
-        let wsv = state.wsv();
-
         match query_request.0 {
             QueryRequest::Query(QueryWithParameters {
                 query,
                 sorting,
                 pagination,
             }) => {
-                let output = query.execute(wsv)?;
-
-                wsv.query_handle().handle_query_output(
-                    output,
-                    NonZeroUsize::new(30_000).expect("30 000 is not zero"),
-                    &sorting,
-                    pagination,
-                )
+                let batched = {
+                    let wsv = state.wsv();
+                    let output = query.execute(wsv)?;
+
+                    wsv.query_handle().handle_query_output(
+                        output,
+                        NonZeroUsize::new(30_000).expect("30 000 is not zero"),
```

Comment:
what is this magic number?

Reply:
It was introduced in my previous PR with live queries for wasm. It's fixed in #4016